runtime/src/main/java/com/redhat/ipaas/rest/ComponentGroups.java

```diff
@@ -37,12 +38,15 @@
     @Inject
     private DataManager dataMgr;
 
+    @Context
+    private UriInfo uri;
+
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = "List component groups")
     @ApiResponses(value = {@ApiResponse(code = 200, message = "Success", response = ComponentGroup.class)})
     public Collection<ComponentGroup> list() {
-        return dataMgr.fetchAll(ComponentGroup.class);
+        return dataMgr.fetchAll(ComponentGroup.class, new ReflectiveSorter<>(ComponentGroup.class, uri.getQueryParameters()));
```

Comment:
It's the responsibility of the rest service to take the incoming raw data (body, query string, cookies, etc) & turn it into a representation that internal services that it calls can use, in this case it's translating the query string parameters for sort into a form that the sorter can use, SortOptions. Why is it a problem to do this in every list operation, with perhaps a method in an introduced superclass so we can get some more consistency or via a simple extractor utility?